azkaban-hadoop-security-plugin/src/main/java/azkaban/security/HadoopSecurityManager_H_2_0.java

```diff
@@ -470,37 +469,47 @@ public void prefetchToken(final File tokenFile, final Props props, final Logger
       throws HadoopSecurityManagerException {
     final String userToProxy = props.getString(JobProperties.USER_TO_PROXY);
 
-    logger.info("Getting hadoop tokens based on props for " + userToProxy);
     doPrefetch(tokenFile, props, logger, userToProxy);
   }
 
   private void doPrefetch(final File tokenFile, final Props props, final Logger logger,
       final String userToProxy) throws HadoopSecurityManagerException {
+    // Create suffix to be added to headless user, the suffix is a valid email ID which conforms
+    // to Kubernetes.
+    final String suffix =
+        props.getBoolean(HadoopSecurityManager.APPEND_SUBMIT_USER, false) &&
+            props.getString(HadoopSecurityManager.SUBMIT_USER_SUFFIX) != null ?
+            "/" + props.getString(Constants.FlowProperties.AZKABAN_FLOW_SUBMIT_USER) +
+                props.getString(HadoopSecurityManager.SUBMIT_USER_SUFFIX) : "";
+
+    final String userToProxyFQN = userToProxy + suffix;
+    logger.info("Getting hadoop tokens based on props for " + userToProxyFQN);
+
     final Credentials cred = new Credentials();
-    fetchMetaStoreToken(props, logger, userToProxy, cred);
-    fetchJHSToken(props, logger, userToProxy, cred);
+    fetchMetaStoreToken(props, logger, userToProxyFQN, cred);
```

Comment:
How would the current `getDelegationToken` work if `userToProxy` is changed?

Reply:
userToProxy would change from something like "azkTest" to "azkTest/deepak@xyz.com"
Delegation tokens work if a valid email ID is provided as tested on actual Azkaban cluster.
The original userToProxy is only used for prepareTokenFile() method as OS user still remains same.